### PR TITLE
protoc-gen-openapiv2: Support all HTTP methods supported in OpenAPI v2

### PR DESCRIPTION
**Pull request by mnito**
#### Have you read the [Contributing Guidelines](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/CONTRIBUTING.md)?
Yep

#### Brief description of what is fixed or changed

The handling for mappings with the same path and method caused a crash when provided with an operation with a HEAD or OPTIONS HTTP method, which are all technically supported by OpenAPI v2.

Also fix this same crash when an unsupported HTTP method is provided

Add HEAD, OPTIONS, and TRACE examples to `a_bit_of_everything.proto`

#### Other comments

I didn't dig into the "why" of the existing pattern too much -- I wonder if it's possible to simplify in a follow-up change to not require the explicit HTTP method switch statements etc?

### CODE REVIEW

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]
Hi, thanks for this fix. I guess no one ever tried using these verbs 😂. Do you mind adding some example methods with these verbs as mappings to our examples? See https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/internal/proto/examplepb/a_bit_of_everything.proto

**Comment by mnito** [comment]
> Hi, thanks for this fix. I guess no one ever tried using these verbs 😂. Do you mind adding some example methods with these verbs as mappings to our examples? See https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/internal/proto/examplepb/a_bit_of_everything.proto

Ha yeah I got a little cute some years back and used HEAD for an exists method in a service and it crashed when upgrading gRPC gateway 😭 . Yeh I can do that.

**Comment by mnito** [comment]
Should be good to go now, let me know what you think

File: protoc-gen-openapiv2/internal/genopenapi/types.go [hunk 1]
```diff
@@ -105,11 +105,15 @@ type openapiPathsObject map[string]openapiPathItemObject
...
-	Patch  *openapiOperationObject `json:"patch,omitempty" yaml:"patch,omitempty"`
+	Get     *openapiOperationObject `json:"get,omitempty" yaml:"get,omitempty"`
+	Delete  *openapiOperationObject `json:"delete,omitempty" yaml:"delete,omitempty"`
+	Post    *openapiOperationObject `json:"post,omitempty" yaml:"post,omitempty"`
+	Put     *openapiOperationObject `json:"put,omitempty" yaml:"put,omitempty"`
+	Patch   *openapiOperationObject `json:"patch,omitempty" yaml:"patch,omitempty"`
+	Head    *openapiOperationObject `json:"head,omitempty" yaml:"head,omitempty"`
+	Options *openapiOperationObject `json:"options,omitempty" yaml:"options,omitempty"`
+	// While TRACE is supported in OpenAPI v3, it is not supported in OpenAPI v2
+	// Trace   *openapiOperationObject `json:"trace,omitempty" yaml:"trace,omitempty"`
```
Comment (johanbrandhorst) [inline protoc-gen-openapiv2/internal/genopenapi/types.go:116]: _nice_

**Review by johanbrandhorst — APPROVED** [review]
This is a quality PR, thanks for adding the integration tests!

**Comment by mnito** [comment]
Thanks!

**Pull request merged**